internal/provider/providers/porkbun/provider.go

```diff
@@ -148,17 +163,31 @@ func (p *Provider) Update(ctx context.Context, client *http.Client, ip netip.Add
 	return ip, nil
 }
 
-func (p *Provider) deleteALIASRecordIfNeeded(ctx context.Context, client *http.Client) (err error) {
-	aliasRecordIDs, err := p.getRecordIDs(ctx, client, "ALIAS")
+// Deletes the matching record for a specific recordType iff the content matches the expected content value.
+// Returns an error if there is a matching record but an unexpected content value.
+// If there is no matching record, there is nothing to do.
+func (p *Provider) deleteMatchingRecord(
+	ctx context.Context,
+	client *http.Client,
+	recordType string,
+	expectedContent string,
+) error {
+	records, err := p.getRecords(ctx, client, recordType)
 	if err != nil {
-		return fmt.Errorf("getting ALIAS record IDs: %w", err)
-	} else if len(aliasRecordIDs) == 0 {
-		return nil
+		return fmt.Errorf("getting %s records: %w", recordType, err)
 	}
 
-	err = p.deleteAliasRecord(ctx, client)
-	if err != nil {
-		return fmt.Errorf("deleting ALIAS record: %w", err)
+	if len(records) == 1 && records[0].Content == expectedContent {
+		err = p.deleteRecord(ctx, client, recordType)
+		if err != nil {
+			return fmt.Errorf("deleting %s record: %w", recordType, err)
+		}
+		return nil
+	} else if len(records) >= 1 {
+		// We have 1 or more matching records, but the expectedContent didn't match. Throw a conflicting record error.
+		return fmt.Errorf("deleting %s record: %w", recordType, errors.ErrConflictingRecord)
 	}
+
+	// No record found for recordType, nothing to do.
 	return nil
```

Comment:
Changed to a switch block for better clarity I hope + changed error wrappings a bit

```suggestion
	switch {
	case len(records) == 0:
		return nil
	case len(records) > 1:
		return fmt.Errorf("%w: %d %s records are already set", errors.ErrConflictingRecord, recordType)
	case records[0].Content != expectedContent:
		return fmt.Errorf("%w: %s record has content %q mismatching expected content %q",
			errors.ErrConflictingRecord, recordType, records[0].Content, expectedContent)
	}
	
	// Single record with content matching expected content.
	err = p.deleteRecord(ctx, client, recordType)
	if err != nil {
		return fmt.Errorf("deleting record: %w", err)
	}
	return nil
```